src/CatalogService/src/Application/Product/Commands/UpdateProduct/UpdateProductCommand.cs

```diff
@@ -1,11 +1,13 @@
-﻿using Catalog.Abstractions;
+﻿using AutoMapper;
+using Catalog.Abstractions;
 using CatalogService.Application.Common.Exceptions;
 using CatalogService.Application.Common.Interfaces;
+using CatalogService.Application.Common.Models;
 using CatalogService.Domain.Events;
 using MediatR;
 
 namespace CatalogService.Application.Product.Commands.UpdateProduct;
-public class UpdateProductCommand : IRequest<int>
+public class UpdateProductCommand : IRequest<ProductDto>

```

Comment:
Returning the model as a result of the command violates CQS. It might be ok in some rare cases, but it's hard to support in future. Please, send another request to the mediator to get a needed model

Reply:
fixed